src/server/accessors/UploadCreator.ts

```diff
@@ -0,0 +1,50 @@
+import * as FileType from 'file-type';
+import * as MIME from 'mime-types';
+
+import { IUploadCreator } from '../../definition/accessors';
+import { IRoom } from '../../definition/rooms';
+import { IUpload } from '../../definition/uploads';
+import { IUploadDetails } from '../../definition/uploads/IUploadDetails';
+import { AppBridges } from '../bridges';
+
+export class UploadCreator implements IUploadCreator {
+    constructor(
+        private readonly bridges: AppBridges,
+        private readonly appId: string,
+    ) { }
+
+    public async createUpload(room: IRoom, filename: string, buffer: Buffer): Promise<IUpload> {
+        const appUser = await this.bridges.getUserBridge().getAppUser(this.appId);
+
+        if (!appUser) {
+            throw new Error('Failed to obtain the app user to perform the upload');
+        }
+
+        if (filename.split('.').length < 2) {
+            throw new Error('The upload filename without extension is not allowed. ' +
+                'Please check the filename of your upload!');
+        }
+
+        const detectedFileType = await FileType.fromBuffer(buffer);
+        const ext = filename.split('.').pop();
+
+        let mime = MIME.lookup(ext) || '';
+        if (detectedFileType) {
```

Comment:
Rocket.Chat should be responsible to validate this aspects of the file, because the admin can configure some validations